src/analytics.js

```diff
@@ -106,3 +111,39 @@ export function insertAnalyticsElement(
   parentElement.appendChild(analyticsElem);
   return analyticsElem;
 }
+
+/**
+ * A function that is used by extension element to get analytics events.
+ * The function makes sure to create or remove sandbox amp-analytics element at right time.
+ * @param {!AmpElement} parent
+ * @param {!Promise<!JsonObject>} promise
```

Comment:
I kept this unchanged, because I want the promise here to be type `{!Promise<!JsonObject>}` but I will need to set it to `promise = null` after unlayout to prevent false adding element if promise resolve after unlayout. It would require me to create an extra param to store the state anyway if I force the input promise to by not null.